client/src/components/ProtectedRoute.tsx

```diff
@@ -0,0 +1,20 @@
+import React, { Component } from 'react';
+import { Route, Redirect } from 'react-router-dom';
+import { useAuth } from '../context/useAuthContext';
+
+const ProtectedRoute = ({ path, component: Component, ...rest }: { path: string; component: any }) => {
+  const loggedInUser = useAuth();
+  return (
+    <Route
+      render={(props) => {
+        if (loggedInUser) {
+          <Component />;
+        } else {
+          return <Redirect to={{ pathname: path, state: { from: props.location } }} />;
```

Comment:
Won't we need to redirect to the path '/login' here?

Reply:
resolved